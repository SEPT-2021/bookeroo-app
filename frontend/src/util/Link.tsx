import React from "react";
import { LinkProps } from "@material-ui/core";
import {
  Link as RouterLink,
  LinkProps as RouterLinkProps,
} from "react-router-dom";

function Link(props: Omit<LinkProps, "component"> & RouterLinkProps) {
  return <Link variant="body2" {...props} component={RouterLink} />;
}

export default Link;
