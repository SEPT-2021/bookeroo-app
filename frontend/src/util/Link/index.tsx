import React from "react";
import { LinkProps, Link as MuiLink } from "@material-ui/core";
import {
  Link as RouterLink,
  LinkProps as RouterLinkProps,
} from "react-router-dom";

function Link(props: Omit<LinkProps, "component"> & RouterLinkProps) {
  return <MuiLink variant="body2" {...props} component={RouterLink} />;
}

export default Link;
