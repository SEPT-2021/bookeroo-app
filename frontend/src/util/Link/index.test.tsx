import React from "react";
import { render } from "@testing-library/react";
import Link from ".";

test("renders internal link", () => {
  const { container } = render(<Link to="/example-page" />);
  expect(container).toMatchSnapshot();
});
