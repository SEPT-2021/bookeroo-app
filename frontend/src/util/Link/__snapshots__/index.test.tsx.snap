// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders internal link 1`] = `
<div>
  <mock-link
    to="/example-page"
  />
</div>
`;
